src/formatDate.js

```diff
@@ -50,7 +50,40 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let result = '';
+
+  const dateObj = {
+    DD: '',
+    MM: '',
+    YY: '',
+    YYYY: '',
+  };
+
+  const dateSplit = date.split(fromFormat[3]);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObj[fromFormat[i]] = dateSplit[i];
+  }
+
+  if (dateObj.YYYY === '' && dateObj.YY >= 30) {
+    dateObj.YYYY = 19 + dateObj.YY;
+  }
+
+  if (dateObj.YYYY === '' && dateObj.YY < 30) {
```

Comment:
Also you can combine your `if` in helper function. This expression `dateObj.YYYY === ''`  is repeated

Reply:
that means that you can create cutYear function and use it in this way:
`if (dateObj.YYY) {
  cutYear(dateObj);
}`

or something like it, to avoid repeating